test/built-ins/RegExp/prototype/Symbol.matchAll/isregexp-called-once.js

```diff
@@ -25,16 +25,32 @@ Object.defineProperty(RegExp.prototype, Symbol.match, {
   }
 });
 
-var count = 0;
+var calls = [];
 var o = {
   get [Symbol.match]() {
-    ++count;
+    calls.push('get @@match');
     return false;
   },
-  flags: "",
+  get flags() {
+    calls.push('get flags');
+    return {
+      toString() {
+        calls.push('flags toString');
+      }
+    };
+  },
 };
 
-RegExp.prototype[Symbol.matchAll].call(o, '1');
+RegExp.prototype[Symbol.matchAll].call(o, {
+  toString() {
+    calls.push('arg toString')
+  }
+});
 
 assert.sameValue(0, internalCount);
-assert.sameValue(1, count);
+
+assert.sameValue(calls.length, 4);
```

Comment:
is there no "assert.deepEqual" that would allow comparing the entire array at a time? that way, failures would be more clear about what's missing.

Reply:
there is a `compareArray` helper but I honestly prefer going this way making it clear I'm expecting the given order and items in sequence.

In any case, there is no easy implementation of a `deepEqual` equivalent, I'd be happy to expand this discussion somewhere else.

Reply:
also, compareArray won't show at what point anything is not equal, just a true or false for comparing array like objects.

Reply:
i'd assume that the error message could be something like "got [a, c] expected [a, b, c]"